glean-core/android/src/main/java/mozilla/telemetry/glean/net/BaseUploader.kt

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.telemetry.glean.net
+
+import android.util.Log
+import androidx.annotation.VisibleForTesting
+import mozilla.telemetry.glean.BuildConfig
+import mozilla.telemetry.glean.config.Configuration
+import org.json.JSONException
+import org.json.JSONObject
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+import java.util.TimeZone
+
+/**
+ * The logic for uploading pings: this leaves the actual upload implementation
+ * to the user-provided delegate.
+ */
+class BaseUploader(d: PingUploader) : PingUploader by d {
+    companion object {
+        private const val LOG_TAG: String = "glean/BaseUploader"
+    }
+
+    /**
+     * Log the contents of a ping to the console, if configured to do so in
+     * [Configuration.logPings].
+     *
+     * @param path the URL path to append to the server address
+     * @param data the serialized text data to send
+     * @param config the Glean configuration object
+     */
+    private fun logPing(path: String, data: String, config: Configuration) {
```

Comment:
nit: Passing the whole configuration has always seemed like overkill here when all we need is the boolean.  In fact, maybe the check to log should be in the doUpload function and wrapped around the call to `logPing` and then this function could do exactly what it says it does.

Reply:
This is a great idea! I just changed that.